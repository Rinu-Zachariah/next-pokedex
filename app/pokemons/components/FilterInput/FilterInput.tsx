'use client';
import { useSearchParams, useRouter } from 'next/navigation';
import React, { useCallback, useRef } from 'react';

interface IProps {
  searchText: string;
}

const FilterInput: React.FC<IProps> = ({searchText}) => {
  const router = useRouter();
  const searchParams = useSearchParams();
  const inputRef = useRef<HTMLInputElement>(null);

  const handleSubmit = useCallback((e: React.FormEvent) => {
    e.preventDefault();
    const name = inputRef.current?.value.trim() || '';
    const params = new URLSearchParams(searchParams.toString());
    
    if (name) {
      params.set('name', name);
    } else {
      params.delete('name');
    }
    
    params.set('page', '1'); 
    router.replace(`?${params.toString()}`);  
  }, [searchParams, router]);
  
  return (
    <form onSubmit={handleSubmit} className="mb-6 flex gap-2">
      <input
        type="text"
        ref={inputRef}
        defaultValue={searchText}
        placeholder="Search by exact name"
        className="flex-1 px-3 py-2 rounded-lg border border-gray-300"
      />
      <button
        type="submit"
        className="px-4 py-2 bg-blue-900 text-white rounded-lg font-semibold"
      >
        Search
      </button>
    </form>
  );
};

export default FilterInput;