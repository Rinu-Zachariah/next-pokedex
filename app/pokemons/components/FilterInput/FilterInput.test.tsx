import React from 'react';
import { render } from '@testing-library/react';
import FilterInput from './FilterInput';

jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    replace: jest.fn(),
    prefetch: jest.fn(),
    refresh: jest.fn(),
    back: jest.fn(),
    forward: jest.fn(),
  }),
  usePathname: () => '/',
  useSearchParams: () => ({
    get: jest.fn(),
    set: jest.fn(),
  }),
}));

describe('FilterInput', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<FilterInput searchText="charizard" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
