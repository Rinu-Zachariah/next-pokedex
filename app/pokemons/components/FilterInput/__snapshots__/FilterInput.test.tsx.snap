// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput renders correctly 1`] = `
<DocumentFragment>
  <form
    class="mb-6 flex gap-2"
  >
    <input
      class="flex-1 px-3 py-2 rounded-lg border border-gray-300"
      placeholder="Search by exact name"
      type="text"
      value="charizard"
    />
    <button
      class="px-4 py-2 bg-blue-900 text-white rounded-lg font-semibold"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
